# **NIVEL BASICO:** ANÁLISIS INTEGRAL

*1. Desarrolla un Web Scraper.*

In [1]:
#Instalacion tabula py
!pip install tabula.py

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 12.0 MB 10.0 MB/s 


In [2]:
#Se importan las librerias
import tabula
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import sklearn as skl
import pickle

In [4]:
#Instalacion great_expectations y boto3
!pip install great_expectations
!pip install boto3

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [5]:
#Importacion de tabula, great_epectations y io para leer PDF
from tabula import read_pdf
import great_expectations as great_expectations
from io import StringIO

*2. Carga los datos extraídos por el web scraper.*

In [8]:
#Convercion del PDF a dataf
ventasc="/content/Venta_Combustibles_2021.pdf"
source_pdf=read_pdf(ventasc,lattice=True, pages="all")
dataf=source_pdf[0]

In [9]:
#Muestra primeras 10 filas de dataf
dataf.head(10)

Ciudad          Estado  Año_mes Tipo_Combustible  \
0           Acapulco        Guerrero   202101        Turbosina   
1     Aguascalientes  Aguascalientes   202101        Turbosina   
2           El Bajio      Guanajuato   202101        Turbosina   
3     Ciudad Obregon          Sonora   202101        Turbosina   
4      Ciudad Juarez       Chihuahua   202101        Turbosina   
5  Ciudad Del Carmen        Campeche   202101        Turbosina   
6             Colima          Colima   202101        Turbosina   
7           Campeche        Campeche   202101        Turbosina   
8           Chetumal    Quintana Roo   202101        Turbosina   
9           Culiacan         Sinaloa   202101        Turbosina   

   Litros_de_co\rmbustible  Servicios\rrealizados  
0                  1150919                    433  
1                  1432481                    398  
2                  3798722                    824  
3                   616464                    208  
4                  1747270                    519  
5                  1362085                    397  
6                   293316                     82  
7                   426467                    123  
8                   364544                    111  
9                  2877051                    742

*3. Al ejecutar tu análisis exploratorio, describe el número de columnas y    filas que encuentras.*

In [10]:
#Esto hace que muestre el numero de filas y columnas que existen en el dataf
print('Número de filas: ', dataf.shape[0])
print('Número de columnas: ', dataf.shape[1])

Número de filas:  45
Número de columnas:  6


*4. Calcula la media, máximo, mínimo y desviación estándar de las columnas   con valores numéricos.*

In [11]:
#Calculamos la media, el maximo, el valor minimo y la desviación estandar de las columnas de VN
dataf.describe()

Año_mes  Litros_de_co\rmbustible  Servicios\rrealizados
count      45.0             4.500000e+01              45.000000
mean   202101.0             5.239269e+06             875.711111
std         0.0             1.554830e+07            1980.936953
min    202101.0             9.490000e+02               2.000000
25%    202101.0             1.191030e+05              82.000000
50%    202101.0             7.828550e+05             244.000000
75%    202101.0             2.600838e+06             662.000000
max    202101.0             9.282059e+07           11656.000000

*5.Limpia los datos de  acuerdo a su utilidad.*

In [12]:
#La siguiente función busca cuales columnas contiene valores nulos
dataf.isna().any()

Ciudad                     False
Estado                     False
Año_mes                    False
Tipo_Combustible           False
Litros_de_co\rmbustible    False
Servicios\rrealizados      False
dtype: bool

In [13]:
#Enseguida se muestra la funcion con la cual se busca y cuenta la cantidad de valores nulos del dataf
dataf.isna().sum()

Ciudad                     0
Estado                     0
Año_mes                    0
Tipo_Combustible           0
Litros_de_co\rmbustible    0
Servicios\rrealizados      0
dtype: int64

# ***-Explica detalladamente  por qué es o no es una  buena idea invertir en el  sector económico del que  recopilaste información.***

Seria buena idea invertir en el anterior sector economico debido a que en la informacion analizada muestra que las ventas son buenas, de igual forma nos pudimos dar cuenta que la media de los servicios realizados es de 875.711111 y es muy buena cantidad.